### PR TITLE
Add swagger for typescript backend

### PR DESCRIPTION
**Pull request by alexguo8**
## Notion ticket link

[Set up Swagger UI for REST APIs](https://www.notion.so/uwblueprintexecs/Task-Board-db95cd7b93f245f78ee85e3a8a6a316d?p=48d82954b93c4fb8a9760a379a0fa45d)

## Implementation description
* Add swagger config file

## Steps to test
1. Start the server locally with docker-compose up --build (currently it works with mongo + auth, other combos can be tested after scrubbing) * Note that you may need to change entityRoutes file to use EntityServiceMg
2. Navigate to http://localhost:5000/api-docs
image

3. Test the login endpoint, then add the accessToken using the Authorize button in the top right
4. Test out the remaining endpoints, similar to Postman

## What should reviewers focus on?
* Do all the endpoints work? Should the wording be updated for any endpoints

## Checklist
- [x] My PR name is descriptive and in imperative tense
- [x] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [x] I have run the appropriate linter(s)
- [x] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

**Review by chenrichard10 — APPROVED** [review]
Tried everything out and it works well! For some reason starter_code_v2 database for mongodb does not seem to work as it gives a uid must not be an empty string error, also my authorization role doesn't currently exist in firebase so I used Sherry's email.

File: backend/typescript/server.ts [hunk 1]
```diff
@@ -2,6 +2,8 @@ import cookieParser from "cookie-parser";
 import cors from "cors";
 import express from "express";
 import * as firebaseAdmin from "firebase-admin";
+import swaggerUi from 'swagger-ui-express';
```
Comment (sherryhli) [inline backend/typescript/server.ts:5]: Small nitpick: can we change to double quotes? Same thing below, could run `yarn fix`.

File: backend/typescript/swagger.yml [hunk 2]
```diff
@@ -0,0 +1,517 @@
...
+                  type: object
+                  required:
+                      - intField, stringField, boolField, enumField, stringArrayField
+                  properties:
+                    intField:
+                      type: integer
+                    stringField:
+                      type: string
+                    stringArrayField:
+                      type: string
```
Comment (sherryhli) [inline backend/typescript/swagger.yml:42]: Should be typed as an array right? Same thing below.
```yml
type: array
items:
  type: string
```

**Comment by sherryhli** [comment]
> Tried everything out and it works well! For some reason starter_code_v2 database for mongodb does not seem to work as it gives a uid must not be an empty string error, also my authorization role doesn't currently exist in firebase so I used Sherry's email.

@chenrichard10 sorry should've announced this but we're not using the `starter-code-v2` DB for testing anymore so the users in there are not up to date. We use either `typescript-test` or `python-test` depending on the backend, you'll just need to change the DB name part of the env variable. Feel free to use my email to test though.

Documented in my README update: https://github.com/uwblueprint/starter-code-v2/tree/sherry/update-readme#set-up

**Review by sherryhli — APPROVED** [review]
LGTM, 🚢 it!

**Pull request merged**